fibonaci/fibonacci.py

```diff
@@ -0,0 +1,122 @@
+from decimal import Decimal, getcontext
+import math
+from functools import lru_cache
+
+__all__ = [
+    "fibonacci_closed",
+    "fibonacci_loop",
+    "fibonacci_range",
+    "fibonaci_loop_max",
+    "fibonacci_recursive",
+]
+
+
+def fibonacci_closed(n: int) -> int:
+    """Function that returns the nth number in the fibonacci sequence
+
+    Args:
+        n (int): Return the number of the fibonacci sequence of this index
+
+    Returns:
+        int: The nth number in the fibonacci sequence
+    """
+    getcontext().prec = 1000
+    phi = Decimal((1 + Decimal(5).sqrt()) / 2)
+    psi = Decimal((1 - Decimal(5).sqrt()) / 2)
+    return int((phi ** n - psi ** n) / Decimal(5).sqrt()+Decimal(0.5))
+
+
+def fibonacci_loop(n: int) -> int:
+    """Function that returns the nth number in the fibonacci sequence
+
+    Args:
+        n (int): Return the number of the fibonacci sequence of this index
+
+    Returns:
+        int: The nth number in the fibonacci sequence
+    """
+    if n < 0:
+        return None
```

Comment:
Je geeft hier `None` terug, terwijl in je als return type alleen `int` hebt aangegeven. 
Je kan hiervoor eens naar `typing.Optional` kijken.

Ik zou hier trouwens echt een fout willen zien als de waarde kleiner is dan 0.